src/System.Management.Automation/CoreCLR/EventResource.cs

```diff
@@ -0,0 +1,616 @@
+#if UNIX
+/*
+    This code was generated by the tools\ResxGen\ResxGen.ps1 run against PowerShell-Core-Instrumentation.man.
+    To add or change logged events and the associated resources, edit PowerShell-Core-Instrumentation.man
+    then rerun ResxGen.ps1 to produce an updated CS and Resx file.
+*/
+using System.Collections.Generic;
+using System.Management.Automation.Internal;
+using System.Runtime.InteropServices;
+
+namespace System.Management.Automation.Tracing
+{
+    /// <summary>
+    /// Provides a class for describing a message resource for an ETW event.
+    /// </summary>
+    internal static class EventResource
+    {
+        // Defines the resource id of the message to use when an event id is not valid.
+        const string MissingEventIdResourceName = "MissingEventIdMessage";
+
+        /// <summary>
+        /// Gets the name of the message resource to use for event ids that are not found.
+        /// is not found.
+        /// </summary>
+        /// <remarks>
+        /// This method is called when GetMessage returns a null value indicating the passed
+        /// in event id was not found. The message should be used as the format string
+        /// with the event id as the single variable argument.
+        /// <remarks>
+        public static string GetMissingEventMessage(out int parameterCount)
+        {
+            parameterCount = 1;
+            return MissingEventIdResourceName;
+        }
+
+        /// <summary>
+        /// Gets the message resource id for the specified event id
+        /// </summary>
+        /// <param name="eventId">The event id for the message resource to retrieve.</param>
+        /// <param name="parameterCount">The number of parameters required by the message resource</param>
+        /// <returns>The string resource id of the associated event message; otherwise, a null reference if the event id is not valid.</returns>
+        public static string GetMessage(int eventId, out int parameterCount)
+        {
+            switch (eventId)
```

Comment:
Make sense replace the gigantic switch with Dictionary?

Reply:
It was a dictionary; PR feedback indicates the switch is much more performant.  Since we want to minimize the overhead of logging, I agreed.

Reply:
Thanks for clarify. I see that C# compiler try to optimize a switch.

Closed.